mne/report.py

```diff
@@ -765,7 +765,7 @@ def _add_figs_to_section(self, figs, captions, section='custom',
             # on some version mayavi.core won't be exposed unless ...
             from mayavi import mlab  # noqa, mlab imported
             import mayavi
-        except ImportError:
+        except:  # on some systems importing Mayavi raises SystemExit (!)
```

Comment:
but this way we would learn it, the way you propose would prevent us from
learning that.

2015-04-29 15:59 GMT+02:00 Eric Larson <notifications@github.com>:

> In mne/report.py
> <https://github.com/mne-tools/mne-python/pull/2038#discussion_r29337681>:
>
> > @@ -765,7 +765,7 @@ def _add_figs_to_section(self, figs, captions, section='custom',
> >              # on some version mayavi.core won't be exposed unless ...
> >              from mayavi import mlab  # noqa, mlab imported
> >              import mayavi
> > -        except ImportError:
> > +        except:  # on some systems importing Mayavi raises SystemExit (!)
>
> ...because apparently we can't be confident what subset of exceptions
> mayavi and its dependencies will throw, unfortunately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2038/files#r29337681>.
>